loopout.go

```diff
@@ -1020,26 +1020,43 @@ func (s *loopOutSwap) waitForConfirmedHtlc(globalCtx context.Context) (
 	return txConf, nil
 }
 
-// waitForHtlcSpendConfirmed waits for the htlc to be spent either by our own
+// waitForHtlcSpendConfirmedV2 waits for the htlc to be spent either by our own
 // sweep or a server revocation tx. During this process, this function will try
 // to spend the htlc every block by calling spendFunc.
 //
 // TODO: Improve retry/fee increase mechanism. Once in the mempool, server can
 // sweep offchain. So we must make sure we sweep successfully before on-chain
 // timeout.
-func (s *loopOutSwap) waitForHtlcSpendConfirmed(globalCtx context.Context,
+func (s *loopOutSwap) waitForHtlcSpendConfirmedV2(globalCtx context.Context,
 	htlcOutpoint wire.OutPoint, htlcValue btcutil.Amount) (
 	*chainntnfs.SpendDetail, error) {
 
+	spendChan := make(chan *chainntnfs.SpendDetail)
+	spendErrChan := make(chan error, 1)
+	quitChan := make(chan bool, 1)
+
+	notifier := sweepbatcher.SpendNotifier{
+		SpendChan:    spendChan,
+		SpendErrChan: spendErrChan,
+		QuitChan:     quitChan,
+	}
+
+	sweepReq := sweepbatcher.SweepRequest{
+		SwapHash: s.hash,
+		Outpoint: htlcOutpoint,
+		Value:    htlcValue,
+		Notifier: notifier,
+	}
+
+	select {
```

Comment:
Would be much cleaner if the sweeper would expose an `AddSweep()` function which takes a request. That we the sweepers channel isn't leaked to the loopout state machine.

Reply:
Doing that would imply using locking logic in the sweeper codebase. The reason I used a channel to deliver everything to it was avoiding locks on all the sweeper methods

Reply:
No need for lock, what I mean is that this is a "leaky abstraction" and ideally these details should be hidden behind a function that is part of the sweeper.

Reply:
bumping.